src/testRunner/unittests/tsserver/compileOnSave.ts

```diff
@@ -951,7 +951,7 @@ function bar() {
                 }
 
                 // Change file1 get affected file list
-                verifyLocalEdit(file1, "hello", "world");
+                verifyLocalEdit(file1, "hello", "world", /*returnsAllFilesAsAffected*/ !declaration); // Signatures are not initialized before this request

```

Comment:
I wonder if compileOnSave should disable this behavior given it doesnt save all files so will be anyways calculating signatures for affected set and not going to change from one edit to next ?

Reply:
@uniqueiniquity @jessetrinity @amcasey thoughts ?

Reply:
@sheetalkamat Can you please elaborate?  I think you're saying
1. Old behavior: compute (emit?) .d.ts on compile-on-save
2. New behavior: don't compute (emit?) .d.ts on (first?) compile-on-save
3. Proposed behavior: restore old behavior

Reply:
If @amcasey's explanation is correct, then yes - my instinct says that compile-on-save _should_ compute+emit .d.ts.

Reply:
Yes thats what i meant.. i will revert to old behavior for compile on save

Reply:
Updated and ready for review